rest_server/src/models/model_init_handler.js

```diff
@@ -46,12 +49,40 @@ const createTemplates = async models => {
               : [item.categories],
           },
         };
-        await models.MarketplaceItem.orm.create(newItem);
+        const marketplaceItem = await models.MarketplaceItem.orm.create(
+          newItem,
+        );
+        if (Array.isArray(newItem.categories)) {
+          for (const categoryName of newItem.categories) {
+            itemCategories.push([categoryName, marketplaceItem.id]);
+          }
+        }
+        if (Array.isArray(newItem.tags)) {
+          for (const tagName of newItem.tags) {
+            itemTags.push([tagName, marketplaceItem.id]);
+          }
+        }
       } catch (err) {
         console.log(err.message);
       }
     }),
   );
+  try {
+    for (const [categoryName, itemId] of itemCategories) {
+      const [category] = await models.ItemCategory.orm.findOrCreate({
+        where: { name: categoryName },
+      });
+      category.addMarketplaceItem(itemId);
+    }
+    for (const [tagName, itemId] of itemTags) {
+      const [tag] = await models.ItemTag.orm.findOrCreate({
+        where: { name: tagName },
+      });
+      tag.addMarketplaceItem(itemId);
+    }
+  } catch (err) {
+    console.log(err.message);
```

Comment:
>console [](http://example.com/codeflow?start=4&length=7)

just curious: if some of the items are added successfully, and then an exception happens, what should we do? we need rollback or just keep the current state?

Reply:
I think it's OK to keep the current state here.